app/models/deferred_offer_confirmation.rb

```diff
@@ -11,7 +11,9 @@ class CourseForm < DeferredOfferConfirmation
     def courses_for_select
       @courses_for_select ||= offer.provider.courses
            .where(recruitment_cycle_year: RecruitmentCycleTimetable.current_year)
+           .joins(:course_options).where('course_options.site_still_valid': true)
```

Comment:
We were allowing courses that didn't have sites / valid sites to show up in the drop down list.